### PR TITLE
ignore_git → omit_git_hash

### LINKED ISSUE #110020: Rename `ignore-git` to `omit-git-hash`
As mentioned by @jyn514 [on Zulip](https://rust-lang.zulipchat.com/#narrow/stream/326414-t-infra.2Fbootstrap/topic/Test.20for.20.23107094/near/347478265), the `ignore-git` option of the bootstrap configuration should be renamed to `omit-git-hash` to better describe its meaning and distinguish it from `submodules = false`. It indicates that information about the current commit shouldn't be displayed when e.g. `rustdoc --version --verbose` is run. This isn't clear from the current name.

### PR DESCRIPTION
**Pull request by ghost**
Fixes #110020

### CODE REVIEW

**Comment by gimbles** [comment]
(It appears I accidentally added a commit from another PR of mine, will fix)

**Review by KittyBorgX — APPROVED** [review]
LGTM 👍

**Review by onur-ozkan — APPROVED** [review]
thanks for the fix!

File: src/doc/unstable-book/src/language-features/lang-items.md
[the remark this answers is not in the document]
Comment (gimbles) [reply src/doc/unstable-book/src/language-features/lang-items.md:1]: Yeah that was a git oops, fixed.

**Pull request merged**